### PR TITLE
Prepare a basic implementation of core functionality #1

### PR DESCRIPTION
**Pull request by mialeska**

### CODE REVIEW

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Applications/IApplication.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using OpenQA.Selenium.Remote;
+
+namespace Aquality.Selenium.Core.Applications
+{
+    public interface IApplication
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Applications/IApplication.cs]: add documentation to public members

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Applications/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using Aquality.Selenium.Core.Waitings;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Reflection;
+
+namespace Aquality.Selenium.Core.Applications
+{
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services, Func<IServiceProvider, IApplication> applicationProvider)
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Applications/Startup.cs]: add documentation to public members

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Configurations/LoggerConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        /// <summary>
+        /// Instantiates class using JSON file with general settings.
+        /// </summary>
+        /// <param name="settingsFile">JSON settings file.</param>
+        public LoggerConfiguration(JsonFile settingsFile)
+        {
+            this.settingsFile = settingsFile;
+        }
+
+        public string Language
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Configurations/LoggerConfiguration.cs]: add documentation to public members

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Constants.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿namespace Aquality.Selenium.Core.Elements
+{
+    public class PopularClassNames
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Constants.cs]: add documentation to public members

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/DesiredState.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    /// </summary>
+    public class DesiredState
+    {
+        public DesiredState(Func<IWebElement, bool> elementStateCondition, string stateName)
+        {
+            ElementStateCondition = elementStateCondition;
+            StateName = stateName;
+        }
+
+        public Func<IWebElement, bool> ElementStateCondition { get; }
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/DesiredState.cs]: add documentation to public members

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Interfaces/IElementFactory.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+using System.Collections.Generic;
+
+namespace Aquality.Selenium.Core.Elements.Interfaces
+{
+    /// <summary>
+    /// Defines the interface used to create the elements.
+    /// </summary>
+    public interface IElementFactory
+    {
+        T GetCustomElement<T>(ElementSupplier<T> elementSupplier, By locator, string name, ElementState state = ElementState.Displayed) where T : IElement;
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Interfaces/IElementFactory.cs]: add documentation to public members

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Utilities/EnumExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Diagnostics.Contracts;
+
+namespace Aquality.Selenium.Core.Utilities
+{
+    public static class EnumExtensions
+    {
+        internal static T ToEnum<T>(this object value) where T : struct, IConvertible
```
Comment (mialeska) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Utilities/EnumExtensions.cs]: should not be internal

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Aquality.Selenium.Core.csproj [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    <None Include="..\..\..\LICENSE">
+      <Pack>True</Pack>
+      <PackagePath></PackagePath>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
+    <PackageReference Include="NLog" Version="4.6.6" />
+    <PackageReference Include="Selenium.Support" Version="3.141.0" />
+    <PackageReference Include="Selenium.WebDriver" Version="3.141.0" />
```
Comment (pavelanihimovsky) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Aquality.Selenium.Core.csproj]: Is it possible to remove dependencies on Selenium from core? In this I have to support selenium version in several places. In this library and in selenium for web library, isn't it?

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Constants.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+    public class Attributes
+    {
+        private Attributes()
+        {
+        }
+
+        public const string Class = "class";
+        public const string Value = "value";
+    }
```
Comment (pavelanihimovsky) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Constants.cs]: Is it core functionality? Is it relevant for mobile?

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Element.cs [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                return (RemoteWebElement)Finder.FindElement(Locator, elementState, timeout);
+            }
+            catch (NoSuchElementException ex)
+            {
+                Logger.Debug($"Page source:{Environment.NewLine}{Application.Driver.PageSource}", ex);
+                throw;
+            }
+        }
+
+        public string GetText()
```
Comment (pavelanihimovsky) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/Element.cs]: Can it be replaced with property?

**Pull request closed without merging**